dev_guide: merge doc related to sanity tests

pilou- opened the pull request:
##### SUMMARY
`dev_guide`: merge `testing/sanity/index.rst` and `testing_sanity.rst`

##### ISSUE TYPE
 - Docs Pull Request

##### COMPONENT NAME
dev_guide

##### ANSIBLE VERSION
```
ansible-playbook 2.5.0 (devel 588bddbc35) last updated 2017/09/08 17:06:00 (GMT +200)
```

dharmabumstead reviewed (approved):
Thanks @pilou-

pilou- commented:
Could this one be merged ?

bcoca commented:
cc @mattclay

mattclay commented:
Instead of continuing to maintain this page, we should migrate any relevant content to https://docs.ansible.com/ansible/devel/dev_guide/testing/sanity/ and associated pages.

The script https://github.com/ansible/ansible/blob/devel/docs/bin/testing_formatter.sh is used to generate that page since it has a dynamic index based on available sanity tests.

pilou- committed:
doc: dev_guide, merge doc related to sanity tests

merge testing/sanity/index.rst and testing_sanity.rst:
- testing/sanity/index.rst wasn't listed in TOC
- remove hardcoded list of sanity tests

pilou- commented:
@mattclay thanks for pointing that! I merged `testing/sanity/index.rst` and `testing_sanity.rst`.

mattclay commented on docs/bin/testing_formatter.sh:
```diff
@@ -1,12 +1,47 @@
 #!/bin/bash -eu
 
-cat <<- EOF > ../docsite/rst/dev_guide/testing/sanity/index.rst
+cat <<- 'EOF' > ../docsite/rst/dev_guide/testing_sanity.rst
```
I think it makes more sense to move the testing_sanity contents to the sanity index.

@dharmabumstead Thoughts?

pilou- replied:
I might have missed something but it seems `sanity/index.rst` wasn't referenced anywhere.

mattclay replied:
I think it should be referenced instead of the old testing_sanity page.

pilou- replied:
Where should be put the reference ?

Note that `dev_guide/testing/compile/index.rst` isn't referenced too and refers to `testing_compile` (this one is referenced).
```
Compile Tests
=============

See :doc:`../../testing_compile` for more information.
```

dharmabumstead replied:
@mattclay that works for me.

dharmabumstead commented:
@mattclay is this good to go?

mattclay commented:
@dharmabumstead The sanity docs still need to be moved to back to `sanity/index.rst`.

dharmabumstead commented:
Closing this one. Feel free to rebase and reopen.
